builtin/mainmenu/dlg_config_world.lua

```diff
@@ -24,159 +24,118 @@ local function modname_valid(name)
-	retval = retval .. pkgmgr.render_packagelist(data.list)
-	retval = retval .. ";" .. data.selected_mod .."]"
-
-	return retval
-end
-
-local function enable_mod(this, toset)
-	local mod = this.data.list:get_list()[this.data.selected_mod]
-
-	if mod.is_game_content then
-		-- game mods can't be enabled or disabled
-	elseif not mod.is_modpack then
-		if toset == nil then
-			mod.enabled = not mod.enabled
-		else
-			mod.enabled = toset
-		end
-	else
-		local list = this.data.list:get_raw_list()
-		for i=1,#list,1 do
-			if list[i].modpack == mod.name then
-				if toset == nil then
-					toset = not list[i].enabled
-				end
-				list[i].enabled = toset
-			end
-		end
-	end
+		"table[5.5,0.75;5.75,6;world_config_modlist;" ..
+		pkgmgr.render_packagelist(data.list) .. ";" .. data.selected_mod .."]"
 end
 
-
 local function handle_buttons(this, fields)
-	if fields["world_config_modlist"] ~= nil then
-		local event = core.explode_table_event(fields["world_config_modlist"])
+	if fields.world_config_modlist then
+		local event = core.explode_table_event(fields.world_config_modlist)
 		this.data.selected_mod = event.row
 		core.settings:set("world_config_selected_mod", event.row)
 
 		if event.type == "DCL" then
-			enable_mod(this)
+			pkgmgr.enable_mod(this)
 		end
 
 		return true
 	end
 
-	if fields["key_enter"] ~= nil then
-		enable_mod(this)
+	if fields.key_enter then
+		pkgmgr.enable_mod(this)
 		return true
 	end
 
-	if fields["cb_mod_enable"] ~= nil then
-		local toset = core.is_yes(fields["cb_mod_enable"])
-		enable_mod(this,toset)
+	if fields.cb_mod_enable ~= nil then
+		pkgmgr.enable_mod(this, core.is_yes(fields.cb_mod_enable))
 		return true
 	end
 
-	if fields["btn_mp_enable"] ~= nil or
-		fields["btn_mp_disable"] then
-		local toset = (fields["btn_mp_enable"] ~= nil)
-		enable_mod(this,toset)
+	if fields.btn_mp_enable ~= nil or
+			fields.btn_mp_disable then
+		pkgmgr.enable_mod(this, fields.btn_mp_enable ~= nil)
 		return true
 	end
 
-	if fields["btn_config_world_save"] then
-		local filename = this.data.worldspec.path ..
-				DIR_DELIM .. "world.mt"
+	if fields.btn_config_world_save then
+		local filename = this.data.worldspec.path .. DIR_DELIM .. "world.mt"
 
 		local worldfile = Settings(filename)
 		local mods = worldfile:to_table()
 
 		local rawlist = this.data.list:get_raw_list()
 
-		local i,mod
-		for i,mod in ipairs(rawlist) do
+		for i = 1, #rawlist do
+			local mod = rawlist[i]
 			if not mod.is_modpack and
 					not mod.is_game_content then
 				if modname_valid(mod.name) then
-					worldfile:set("load_mod_"..mod.name, tostring(mod.enabled))
-				else
-					if mod.enabled then
-						gamedata.errormessage = fgettext_ne("Failed to enable mod \"$1\" as it contains disallowed characters. Only chararacters [a-z0-9_] are allowed.", mod.name)
-					end
+					worldfile:set("load_mod_" .. mod.name,
+							tostring(mod.enabled))
+				elseif mod.enabled then
+					gamedata.errormessage = fgettext_ne("Failed to enable mo" ..
```

Comment:
The gettext reading script for our source files will probably fail to read this correctly. Needs testing.